### PR TITLE
Pnp access contr settings

### PR DESCRIPTION
**Pull request by ThorstenLoeschmann**

### CODE REVIEW

**Comment by ThorstenLoeschmann** [comment]
Something seems to be off when loading "Microsoft.Online.SharePoint.PowerShell" Microsoft.SharePoint.Administration.DesignPackageType' from assembly 'Microsoft.SharePoint.Client, Version=16.0.0.0, Culture=neutral, PublicKeyToken=71e9bce111e9429c'

File: Modules/Office365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.psm1 [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+        StartASiteFormUrl            = $null
+        IPAddressEnforcement         = $null
+        IPAddressAllowList           = $null
+        IPAddressWACTokenLifetime    = $null
+        CommentsOnSitePagesDisabled  = $null
+        SocialBarOnSitePagesDisabled = $null
+        DisallowInfectedFileDownload = $null
+        ExternalServicesEnabled      = $null
+        EmailAttestationRequired     = $null
+        EmailAttestationReAuthDays   = $null
```
Comment (NikCharlebois) [inline Modules/Office365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.psm1]: Need to also return GlobalAdminAccount and CentralAdminUrl from the PSBoundParameters directly

File: Modules/Office365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.psm1 [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+            StartASiteFormUrl            = $SPOAccessControlSettings.StartASiteFormUrl
+            IPAddressEnforcement         = $SPOAccessControlSettings.IPAddressEnforcement
+            IPAddressAllowList           = $SPOAccessControlSettings.IPAddressAllowList
+            IPAddressWACTokenLifetime    = $SPOAccessControlSettings.IPAddressWACTokenLifetime
+            CommentsOnSitePagesDisabled  = $SPOAccessControlSettings.CommentsOnSitePagesDisabled
+            SocialBarOnSitePagesDisabled = $SPOAccessControlSettings.SocialBarOnSitePagesDisabled
+            DisallowInfectedFileDownload = $SPOAccessControlSettings.DisallowInfectedFileDownload
+            ExternalServicesEnabled      = $SPOAccessControlSettings.ExternalServicesEnabled
+            EmailAttestationRequired     = $SPOAccessControlSettings.EmailAttestationRequired
+            EmailAttestationReAuthDays   = $SPOAccessControlSettings.EmailAttestationReAuthDays
```
Comment (NikCharlebois) [inline Modules/Office365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.psm1]: Same as above, beed to return GlobalAdminAccount and CentralAdminUrl

File: Modules/Office365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.psm1 [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    Test-PnPOnlineConnection -SPOCentralAdminUrl $CentralAdminUrl -GlobalAdminAccount $GlobalAdminAccount
+
+    $CurrentParameters = $PSBoundParameters
+    $CurrentParameters.Remove("CentralAdminUrl")
+    $CurrentParameters.Remove("GlobalAdminAccount")
+    $CurrentParameters.Remove("Verbose")
```
Comment (NikCharlebois) [inline Modules/Office365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.psm1]: Verbose would never be part of the PSBoundParameters. Therefore it is safe to remove this line

File: Modules/Office365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.psm1 [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [ValidateSet('Yes')]
+        [String]
+        $IsSingleInstance,
+
+        [Parameter()]
+        [System.Boolean]
+        $DisplayStartASiteOption,
```
Comment (NikCharlebois) [inline Modules/Office365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.psm1]: Only the mandatory parameters are required in this signature. To be consistent with other resources please only include IsSingleInstance, GlobalAdminAccount and CentralAdminUrl

File: Modules/Office365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.schema.mof [hunk 5]
```diff
@@ -0,0 +1,18 @@
+[ClassVersion("1.0.0.0"), FriendlyName("SPOAccessControlSettings")]
+class MSFT_SPOAccessControlSettings : OMI_BaseResource
+{
+    [Key, Description("Specifies the resource is a single instance, the value must be 'Yes'"),ValueMap{"Yes"},Values{"Yes"}] String IsSingleInstance;
+    [Write, Description("Determines whether tenant users see the Start a Site menu option")] boolean DisplayStartASiteOption;
+    [Write, Description("Specifies URL of the form to load in the Start a Site dialog.The valid values are:<emptyString> (default) - Blank by default, this will also remove or clear any value that has been set.Full URL - Example: https://contoso.sharepoint.com/path/to/form")] string StartASiteFormUrl;
```
Comment (NikCharlebois) [inline Modules/Office365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.schema.mof]: Add a dpace between the period that ends sentence and new sentence

File: Modules/Office365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.schema.mof [hunk 6]
```diff
@@ -0,0 +1,18 @@
+[ClassVersion("1.0.0.0"), FriendlyName("SPOAccessControlSettings")]
+class MSFT_SPOAccessControlSettings : OMI_BaseResource
+{
```
Comment (NikCharlebois) [inline Modules/Office365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.schema.mof]: The descriptions for the params now automatically drive the wiki documentation. Please make sure each sentence finishes with a ‘.’

File: Modules/Office365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.schema.mof [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    [Key, Description("Specifies the resource is a single instance, the value must be 'Yes'"),ValueMap{"Yes"},Values{"Yes"}] String IsSingleInstance;
+    [Write, Description("Determines whether tenant users see the Start a Site menu option")] boolean DisplayStartASiteOption;
+    [Write, Description("Specifies URL of the form to load in the Start a Site dialog.The valid values are:<emptyString> (default) - Blank by default, this will also remove or clear any value that has been set.Full URL - Example: https://contoso.sharepoint.com/path/to/form")] string StartASiteFormUrl;
+    [Write, Description("Allows access from network locations that are defined by an administrator.")] boolean IPAddressEnforcement;
+    [Write, Description("Configures multiple IP addresses or IP address ranges (IPv4 or IPv6).Use commas to separate multiple IP addresses or IP address ranges.")] string IPAddressAllowList;
```
Comment (NikCharlebois) [inline Modules/Office365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.schema.mof]: Add space between sentences

File: Modules/Office365DSC/Examples/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.ps1 [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    {
+        SPOAccessControlSettings MyTenantAccessControlSettings
+        {
+            IsSingleInstance             = "Yes"
+            CentralAdminUrl              = "https://o365dsc1-admin.sharepoint.com"
+            GlobalAdminAccount           = $credsGlobalAdmin
+            DisplayStartASiteOption      = $false
+            StartASiteFormUrl            = "https://o365dsc1.sharepoint.com"
+            IPAddressEnforcement         = $false
+            #IPAddressAllowList           = "" #would generate an error while writing this resource
```
Comment (NikCharlebois) [inline Modules/Office365DSC/Examples/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.ps1]: Should we simply remove it from the example then?

Comment (ThorstenLoeschmann) [reply Modules/Office365DSC/Examples/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.ps1]: Agreed. I removed it and we can add it back in once it is working.

**Comment by NikCharlebois** [comment]
LGTM

**Pull request merged**